src/main/java/Deadline.java

```diff
@@ -0,0 +1,59 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Deadline class that extends from Task parent class.
+ * Represents a deadline task.
+ */
+public class Deadline extends Task {
+
+    private LocalDate deadline;
+
+    /**
+     * Constructor method.
+     * @param description task description.
+     * @param deadline LocalDate object that describes the deadline.
+     */
+    public Deadline(String description, LocalDate deadline) {
+        super(description);
+        this.deadline = deadline;
+    }
+
+    /**
+     * Overloaded constructor method
+     * @param description task description.
+     * @param deadline LocalDate object that describes the deadline.
+     * @param isDone task done status.
+     */
+    public Deadline(String description, LocalDate deadline, boolean isDone) {
+        super(description, isDone);
+        this.deadline = deadline;
+    }
+
+    /**
+     * Getter method for task deadline.
+     * @return string of task deadline
+     */
+    public LocalDate getDeadline() {
+        return this.deadline;
```

Comment:
Should the redundant `this` be removed?

Reply:
I do feel that `this` is not redundant as it can help with the readability of the code, specifying to return this deadline of the Deadline task.